shell/platform/darwin/macos/framework/Source/FlutterViewController_Internal.h

```diff
@@ -2,20 +2,63 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#include <map>
+#include <unordered_set>
 
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformViews.h"
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterView.h"
 
 @interface FlutterViewController ()
 
 // The FlutterView for this view controller.
 @property(nonatomic, readonly, nullable) FlutterView* flutterView;
 
+// NSDictionary maps strings to FlutterPlatformViewFactorys.
+@property(nonnull, nonatomic)
+    NSMutableDictionary<NSString*, NSObject<FlutterPlatformViewFactory>*>* factories;
+
+// A map of platform view ids to views.
+@property(nonatomic) std::map<int, NSView*> platformViews;
+
+// View ids that are going to be disposed on the next present call.
+@property(nonatomic) std::unordered_set<int64_t> platformViewsToDispose;
+
 /**
  * This just returns the NSPasteboard so that it can be mocked in the tests.
  */
 @property(nonatomic, readonly, nonnull) NSPasteboard* pasteboard;
 
+/**
+ * Platform View Methods.
+ */
+
+/**
+ * Creates a platform view using the arguments from the provided call.
+ * The call's arguments should be castable to an NSMutableDictionary<NSString*, id>*
+ * and the dictionary should at least hold one key for "id" that maps to the view id and
+ * one key for "viewType" which maps to the view type (string) that was used to register
+ * the factory.
+ * FlutterResult is updated to contain nil for success or to contain
+ * a FlutterError if there is an error.
+ */
+- (void)onCreate:(nonnull FlutterMethodCall*)call result:(nonnull FlutterResult)result;
```

Comment:
parse the args in the caller, the methods here would be better phrased as:

`onCreatePlatformViewWithId:() etc`.

Reply:
Done